libr/core/disasm.c

```diff
@@ -3773,7 +3773,19 @@ static int myregwrite(RAnalEsil *esil, const char *name, ut64 *val) {
 		R_FREE (type);
 		if ((ds->printed_flag_addr == UT64_MAX || *val != ds->printed_flag_addr)
 		    && (ds->show_emu_strflag || !emu_str_printed)) {
-			RFlagItem *fi = r_flag_get_i (esil->anal->flb.f, *val);
+			const RList *flags = r_flag_get_list (esil->anal->flb.f, *val);
```

Comment:
this is pretty slow solution, also we dont want such code inside this function, move it into a separate function please.

Reply:
okay, hmm might be better to have a r_skiplist_find_eq function just for r_flag_get_nearest_list(_, _, dir = 0)

Reply:
or maybe not

Reply:
yeah thats a bit hairy